[helm] Document helm migration from `loki-distributed` (#7514)
grafana/loki: docs/sources/installation/helm/migrate-from-distributed/index.md

```yaml
- source_labels:
    - "__meta_kubernetes_pod_label_app_kubernetes_io_component"
  regex: "(canary|read|write)"
  action: "drop"
```

This leverages the fact the the new deployment adds a `app.kubernetes.io/component` label of either `read` for the Read pods, `write` for the Write pods, and `canary` for the Loki Canary pods. These annotations are not present in the `loki-distributed` deployment, so this should only match logs from the new deployment.

### Deploy new Loki Cluster

Next you will deploy the new `loki` chart in the same namespace as your existing `loki-distributed` installation. Make sure to use the same buckets and storage configuration as your existing installation. You will need to set some special `migrate` values as well:

```yaml
migrate:
  fromDistributed:
    enabled: true
    memberlistService: loki-loki-distributed-memberlist
```

> The value of `memberlistService` must be the kubernetes service created for the purpose of Memberlist DNS in the `loki-distributed` deployment. It should match the value of `memberlist.join_members` in the config of the `loki-distributed` deployment. This is what will make the new cluster join the existing clusters ring. It is important to join all existing rings, if you are using different memberlist DNS for different rings, you must manually set the value of each applicable `join_members` configuration for each ring. If using the same memberlist DNS for all rings, as is the default in the `loki-distributed` chart, setting `migrate.memberlistService` should be enough.

Reviewer: ```suggestion
The value of `memberlistService` must be the Kubernetes service created for the purpose of Memberlist DNS in the `loki-distributed` deployment. It should match the value of `memberlist.join_members` in the config of the `loki-distributed` deployment. This is what will make the new cluster join the existing clusters ring. It is important to join all existing rings, if you are using different memberlist DNS for different rings, you must manually set the value of each applicable `join_members` configuration for each ring. If using the same memberlist DNS for all rings, as is the default in the `loki-distributed` chart, setting `migrate.memberlistService` should be enough.
```

  * Logs with the label `job="loki/loki-read"`. The read component does not exist in `loki-distributed`, so this show the new Loki cluster's self monitoring is working correctly.
* Confirm new logs are in the old deployment. Using the old deployment's Loki datasource in Grafana, look for:
  * Logs with the label `job="loki/loki-read"`. Since you have excluded logs from the new deployment from going to the `loki-distributed` deployment, if you can query them through the `loki-distributed` Loki datasource that show the ingesters have joined the same ring, and are queryable from the `loki-distributed` queriers.

### Convert all Clients to Push Logs to New `loki` Deployment

Assuming everything is working as expected, you can now modify the `clients` section of your Grafana Agent or Promtail configuration to push logs to the new deployment. After this change is made, the `loki-distributed` cluster will no longer recieve new logs and can be carefully scaled down.

> Once this has deployed, query the new `loki` cluster's Loki datasource for new logs to make sure they're still being ingested.

Reviewer: ```suggestion
Once this has deployed, query the new `loki` cluster's Loki data source for new logs to make sure they're still being ingested.
```

### Tear Down the Old Loki Canary

> If you had previously deployed the canary via the `loki-canary` helm chart, you can now tear it down. The new chart ships the canary by default and is automatically configured to scrape it.

Reviewer: ```suggestion
If you had previously deployed the canary via the `loki-canary` Helm Chart, you can now tear it down. The new chart ships the canary by default and is automatically configured to scrape it.
```

### Update Flush Config On `loki-distributed` Deployment

You are almost ready to start scaling down the old `loki-distributed` cluster. Before you start, however, it is important to make sure the `loki-distributed` ingesters are configured to flush on shutdown. Since these ingesters will not be coming back, there will be no opportunity for them to replay their WALs, so they need to flush all in-memory logs before shutting down to prevent data loss. 

The easiest way to do this is via the `extraArgs` argument in the `ingester` section of the `loki-distributed` chart. You may also want to set the ingester's log level to `debug` to see messages about the flushing process.

```yaml
ingester:
  replicas: 3
  extraArgs:
    - '-ingester.flush-on-shutdown=true' 
    - '-log.level=debug'
    ```

Deploy this change, and make sure all ingesters restart and are running the latest configuration.

> ### Scale Down Ingesters 1 at a Time

Reviewer: ```suggestion
### Scale Down Ingesters one at a Time
```

> It is now time to start scaling down `loki-distributed`. Scale down the ingester StatefulSet or Deployment (depending on how your `loki-distributed` chart is deployed) 1 replica at a time. If `debug` logs were enabled, you can monitor the logs of each ingester as it's terminating to make sure the flushing process was successful. Once the ingester pod is fully terminated, continue decrementing by another 1 replica. Continue until there are 0 instances of the ingester running.

Reviewer: How is this done? Should we provide `kubectl` commands?

### Remove `loki-distributed` cluster

Double check that logs are still coming in to the new cluster. If something is wrong, it will be much easier to quickly scale back up `loki-distributed` ingesters before tearing down the whole cluster so you can investigate. If everything looks good, you can tear down `loki-distributed` using `helm uninstall`. For example:

```bash
helm uninstall -n loki loki-distributed
```

> Now edit the new `loki` cluster to remove the `migrate` options you added when first installing. Remove all of the following from your `values.yaml`:

Reviewer: What `helm` command is used to apply the changes?
